### DISCUSSION TITLE
OneWire vereinfachen: Alle (und beliebig viele) Dallas-Sensoren auf einen Bus

### DISCUSSION THREAD

**Discussion by rschiemann1** [Zeppelin500/MBusino#3] [Ideas]
Es wurde diskutiert, ob es nicht sinnvoll wäre, alle Dallas-Sensoren auf *einen* oneWire-Bus zu legen, anstatt wie bisher für jeden Dallas-Sensor einen eigenen Anzulegen. In der aktuellen Lösung ist die Anzahl der Dallas-Sensoren limitiert durch die Anzahl der nutzbaren GPIOs vom Wemos Mini.

Hier meine Gedanken zur Vorgehensweise, nur einen oneWire-Bus zu haben:

- beliebig viele Sensoren möglich
- Nutze Methode 1 von [hier](https://lastminuteengineers.com/multiple-ds18b20-arduino-tutorial/). Einmal beim booten vom Wemos wird der Bus nach Sensoren abgesucht. 
- von jedem gefundenen Sensor wird die Adresse abgefragt und daraus ein ID-Stringer erstellt. Dieser String ist der eindeutige Name dieses Sensors (zB auch in MQTT)
- Damit ist es unerheblich, ob beim Absuchen des oneWire-Buses die Sensoren immer in der gleichen Reihenfolge gefunden werden, Eindeutigkeit wird durch die IDs erreicht.
- Der Nutzer muss dann lediglich wissen / herausfinden, welche Sensor-ID (also Adresse) zu welchem Sensor gehört. Das kann er vor dem Sensor-Einbau mit einem einfachen Arduino-Sketch herausfinden, oder er stöpselt neue Sensoren einer nach dem anderen ein und schaut dabei, welche IDs hinzukommen, oder er schließt alle Sensoren gleichzeitig und ordnet die Sensoren anhand der Messungen zu (zB Sensor in warme Hand nehmen)
- Durch diese Änderungen wird das Routing auf der Platine viel einfacher. 
- Wem die Anzahl der Schraubklemmen auf dem Board nicht ausreicht, der kann ja immernoch eine kleine Erweiterungsplatine an eine Schraubklemme anschließen, die wiederrum auf beliebig viele (innerhalb der Limiterungen des oneWire-Buses) Schraubklemmen verteilen kann.

**Comment by Zeppelin500** [comment]
Ich finde die Idee Super! Auch glaube ich, dass wir das hinbekommen.
Was mich noch bremst ist die verlorene Einfachheit, sowohl des Codes als auch der Anwendung.
Nicht jeder ist in der Lage damit umzugehen. Es gibt User, die werden damit nicht zurecht kommen.
Jetzt ist es halt einfach: S1 kommt auch so im MQTT an und mit der .jaml landet es ohne irgend ein Wissen über die Funktionsweise auch im Home-Assistent.
Der Vorteil mit den frei werdenden GPIOs ist auch nicht so groß, eher im Gegenteil. Denn die GPIOs sind jetzt einzeln, zusammen mit der Spannung nach aussen geführt. Wer also einen GPIO für was anders nutzen will kann es jetzt rein über die Veränderung des Codes tun. Er muss nur den Pullup nicht bestücken.
Also, am besten wäre beides. Hab mir schon überlegt ob man über Dipschalter bzw. Lötbrücken wählen kann ob die einzelnen Klemmen über GPIO oder durchverbunden angebunden sind.
